internal/pipe/snapcraft/snapcraft.go

```diff
@@ -328,7 +328,7 @@ func create(ctx *context.Context, snap config.Snapcraft, arch string, binaries [
 			WithField("dst", destBinaryPath).
 			Debug("copying")
 
-		if err = gio.CopyWithMode(binary.Path, destBinaryPath, 0o555); err != nil {
+		if err = gio.CopyWithMode(binary.Path, destBinaryPath, 0o777); err != nil {
```

Comment:
shouldn't it be 0o755?

Reply:
The user's umask setting should handle clearing the group and world writable bits.

Reply:
I agree, it's really weird. I wonder if there is a change in one of the build steps that overwrites the binary now but didn't before? If this is the case then previously the non-writeable permission did not matter, but now it does.

Reply:
no, there shouldn't be any change like that 

i wonder if maybe something changed on snapcraft? maybe it accepted it before but don't accept anymore? 🤔

Reply:
I don't suspect any change in snapcraft. I tried a few tests all on the same machine with snapcraft 8.7.2:

goreleaser v2.7.0: works
goreleaser v2.8.0: permission error
goreleaser v2.8.0 + this PR: works

Reply:
ok, lemme try and investigate a bit more

Reply:
also, as another datapoint, goreleaser build itself created the snap successfully 🤔

Reply:
OK, I'm now totally confused because https://github.com/twpayne/chezmoi/actions/runs/13842252044/job/38732336478?pr=4344 passed with an unmodified v2.8.0.

It's really strange because I got the same permission error on my local Ubuntu development machine and on GitHub Actions. Maybe it was a temporary snapcraft bug. I have no other explanation.

Feel free to close this.

Reply:
weird... yeah I couldn't repro it here either, so, closing.

if it happens again I'll investigate it further 🙏🏻

Reply:
thanks for letting me know about this btw 🙏